drivers/fxos8700/fxos8700.c

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (C) 2018 UC Berkeley
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @file
+ * @brief       Driver for the FXOS8700 3-axis accelerometer/magnetometer
+ *
+ * @author      Michael Andersen <m.andersen@cs.berkeley.edu>
+ * @author      Hyung-Sin Kim <hs.kim@cs.berkeleyedu>
+ */
+
+
+#include "periph/i2c.h"
+#include "xtimer.h"
+#include "fxos8700.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#define I2C_SPEED               I2C_SPEED_FAST
+
+#ifndef FXOS8700_RENEW_INTERVAL
+#define FXOS8700_RENEW_INTERVAL 1000000ul
+#endif
+
+#define ACCEL_ONLY_MODE         (0x00)
+#define MAG_ONLY_MODE           (0x01)
+#define HYBRID_MODE             (0x03)
+
+static fxos8700_measurement_t acc_cached, mag_cached;
```

Comment:
Instead of those static variable; why no having them as attributes of `fxos8700_t` ?
Because otherwise this will be a problem with multiple instances of this device as already reported by @gebart [here](https://github.com/RIOT-OS/RIOT/pull/8978#issuecomment-382612342).That could be a best temporary solution until we have a module for managing cache.
Maybe saul could do it ?

Reply:
I agree and updated this PR reflecting this comment. Thanks!